centreon/www/front_src/src/ApiTokens/TokenListing/Actions/Activate/Activate.tsx

```diff
@@ -12,42 +12,29 @@ import { Row } from '../../models';
 const Activate = ({ row }: Row): React.JSX.Element => {
   const queryClient = useQueryClient();
 
-  const [isPending, startTransition] = useTransition();
   const [isRevoked, setIsRevoked] = useState<boolean>(row?.isRevoked);
 
-  useEffect(() => {
-    if (row?.isRevoked !== isRevoked) {
-      setIsRevoked(row?.isRevoked);
-    }
-  }, [row?.isRevoked]);
-
   const { mutateAsync } = useMutationQuery({
     getEndpoint: () =>
       patchTokenEndpoint({ tokenName: row?.name, userId: row?.user.id }),
     method: Method.PATCH,
+    onError: () => setIsRevoked(!isRevoked),
     onSuccess: () => queryClient.invalidateQueries({ queryKey: ['listTokens'] })
   });
 
   const onClick = (e: React.BaseSyntheticEvent): void => {
-    const value = e.target.checked;
-    startTransition(() => {
-      setIsRevoked(value);
-    });
+    const value = !e.target.checked;
+    setIsRevoked(value);
 
     mutateAsync({
-      payload: { is_revoked: !value }
-    }).then((response) => {
-      if ((response as ResponseError).isError) {
-        setIsRevoked(value);
-      }
+      payload: { is_revoked: value }
     });
   };
 
   return (
     <Switch
       aria-label={labelActiveOrRevoked}
       checked={!isRevoked}
-      disabled={isPending}
```

Comment:
for my point of view the disabled should be , while the request it's hapening

Reply:
outdated code

Reply:
still , the current code there is no disabled for the switch